core/src/zisk_definitions.rs

```diff
@@ -156,3 +155,40 @@ pub const REG_T3: u64 = REG_X28; // Temporary register 3
 pub const REG_T4: u64 = REG_X29; // Temporary register 4
 pub const REG_T5: u64 = REG_X30; // Temporary register 5
 pub const REG_T6: u64 = REG_X31; // Temporary register 6
+
+/// Power of 2 constant definitions, named P2_n, equivalent to 2 to the power of n, in u64 format
+pub const P2_0: u64 = 0x1;
+pub const P2_1: u64 = 0x2;
+pub const P2_2: u64 = 0x4;
+pub const P2_3: u64 = 0x8;
+pub const P2_4: u64 = 0x10;
+pub const P2_5: u64 = 0x20;
+pub const P2_6: u64 = 0x40;
+pub const P2_7: u64 = 0x80;
+pub const P2_8: u64 = 0x100;
+pub const P2_9: u64 = 0x200;
+pub const P2_10: u64 = 0x400;
+pub const P2_11: u64 = 0x800;
+pub const P2_12: u64 = 0x1000;
+pub const P2_13: u64 = 0x2000;
+pub const P2_14: u64 = 0x4000;
+pub const P2_15: u64 = 0x8000;
+pub const P2_16: u64 = 0x10000;
+pub const P2_17: u64 = 0x20000;
+pub const P2_18: u64 = 0x40000;
+pub const P2_19: u64 = 0x80000;
+pub const P2_20: u64 = 0x100000;
+pub const P2_21: u64 = 0x200000;
+pub const P2_22: u64 = 0x400000;
+pub const P2_23: u64 = 0x800000;
+pub const P2_24: u64 = 0x1000000;
+pub const P2_25: u64 = 0x2000000;
+pub const P2_26: u64 = 0x4000000;
+pub const P2_27: u64 = 0x8000000;
+pub const P2_28: u64 = 0x10000000;
+pub const P2_29: u64 = 0x20000000;
+pub const P2_30: u64 = 0x40000000;
+pub const P2_31: u64 = 0x80000000;
```

Comment:
this would be a really good application for a const fn that just calculates the power of 2 number. Since it would be a const fn, it would execute at compile-time, removing the need for the constants

Reply:
Considered as a future improvement